### PR TITLE
Common Attr expression generation to support AND, NOT, OR

### PR DESCRIPTION
**Pull request by borgstrom**
This PR is (sort of) a continuation of #17 -- it implements support for all of the remaining update conditions that were left as `TODO`.

### CODE REVIEW

**Comment by borgstrom** [comment]
@moxizhou -- Can you please review what I added to the docs and let me know if you feel we should expand/improve in any way.

**Comment by moxizhou** [comment]
Looks good I can't think of other supports, but I will let you know when I use any of them. Will probably need the count soon actually

**Comment by borgstrom** [comment]
Just realized I still need to add docs for the update functions (list_append, plus, etc)

**Comment by borgstrom** [comment]
So.... turns out that I couldn't see the forrest from the trees after implementing the logic for `list_append` et al.

I totally didn't need to hand roll my own condition parsing logic as boto3 already provides ALL of that (http://boto3.readthedocs.io/en/latest/reference/customizations/dynamodb.html#boto3.dynamodb.conditions.Attr) and embarrassingly I already wrote an implementation that maps kwarg "double-under" syntax to `Attr`s for the `scan` method... oops! 🤣

I just pushed up 1dff131 that generalizes the mapping to Attr via double under logic and also resolves a long standing TODO to support OR (`|`) and NOT (`~`) queries by taking a page from Django's design.

Now there is a new `Q` function that is responsible for turning keyword arguments into boto3 `Attr` objects that have been `AND`d together.  When you pass keyword args into `scan` or `update` conditions they are directly passed into `Q`.  However, now you can also pass a list of positional arguments into `scan` and `update` conditions, in which case we `AND` together that list.

I'm much happier about where this has ended up and will write some new tests to cover the new Q function explicitly.

File: tests/test_table.py [hunk 1]
```diff
@@ -260,7 +262,7 @@ def test_update_invalid_fields(TestModel, TestModel_entries, dynamo_local):
             unknown_attr='foo'
         )
 
-    with pytest.raises(InvalidSchemaField):
+    with pytest.raises(ConditionFailed):
```
Comment (borgstrom) [inline tests/test_table.py]: I would like to have kept this as an `InvalidSchemaField` from a semantic view point, but now that `Q` is a standalone thing that's not part of a table object it doesn't have access to a schema anymore and technically the condition is failing -- so I'm OK with the change here.

**Comment by moxizhou** [comment]
Oh interesting...hmmm well that is good to know :)

**Pull request merged**